apps/src/dance/danceRedux.ts

```diff
@@ -150,6 +152,9 @@ const danceSlice = createSlice({
     setCurrentSongMetadata: (state, action: PayloadAction<SongMetadata>) => {
       state.currentSongMetadata = action.payload;
     },
+    setShowingAi: (state, action: PayloadAction<object | undefined>) => {
```

Comment:
And here rename as `setShowingAiModal`?

Reply:
Also curious why you chose an object instead of a boolean.

Reply:
Good question.  This is a reference to the field that was clicked to invoke the modal, so that not only do we know that a modal is showing, but we know which field, so that we can set the value on that field when the AI gives us one.  I will rename this to something clearer, maybe `currentAiModalField`.

Reply:
That makes sense! Thanks for the explanation.